### PR TITLE
feat: Parse backticks as html code elements

### PR DESCRIPTION
**Pull request by NisanthanNanthakumar**
## Objective
All code formatted strings are formatted properly.

Naive parsing function should be O(n). **Can this be improved?**

## UI
*BEFORE*
Screen Shot 2020-09-25 at 1 01 23 PM

*AFTER*
Screen Shot 2020-09-25 at 1 03 05 PM

### CODE REVIEW

File: src/templates/apiPage.tsx [hunk 1]
```diff
@@ -13,6 +13,37 @@ import {
 
 import "prismjs/components/prism-json";
 
+const parseBackticks = (str: string) => {
+  let arr = str.split("");
+  let i = 0;
+  let j = arr.length - 1;
+
+  const forward = () => {
+    for (i <= j; i++; ) {
```
Comment (mgaeta) [inline src/templates/apiPage.tsx:22]:
```
for (; i <= j; i++) {
```

**Review by mgaeta — CHANGES_REQUESTED** [review]
`parseBackticks` needs tests.

**Comment by mgaeta** [comment]
This will fail with:
```
`foo` `foo`
```

**Comment by mgaeta** [comment]
```
const parseBackticks = (str) => {
  let i = 0;
  return str.split("").map((c) => {
    return c === "`" ? (i++ % 2 ? "</code>": "<code>") : c;
  }).join("");
};
```

**Comment by NisanthanNanthakumar** [comment]
> const parseBackticks = (str) => {
>   let i = 0;
>   return str.split("").map((c) => {
>     return c === "`" ? (i++ % 2 ? "</code>": "<code>") : c;
>   }).join("");
> };

lol I didn't realize it would break for consecutive, this a good one!!

**Comment by manuzope** [comment]
Hmmm, I don't know if docs even have tests 🤔

**Comment by NisanthanNanthakumar** [comment]
> Hmmm, I don't know if docs even have tests 🤔

Not alot, but the structure is there

**Review by mgaeta — APPROVED** [review]
👍

**Review by mgaeta — APPROVED** [review]
Ok this looks good but there is one more edge case: `` ` `` (a code block containing a single backtick). I don't think we'll actually see that in the docs but I think you should add a comment describing why we squash backticks and why this case isn't covered.
